javascripts/2048.js

```diff
@@ -1,32 +1,112 @@
+var rows = ['0', '1', '2', '3'];
+var cols = ['0', '1', '2', '3'];
+var boardSize = rows.length * cols.length;
+var newTileValue = [ 2, 2, 2, 2, 4 ];
+var score = 0;
+var priorWin = false;
+var gameOver = false;
+
+// win var is currently set to win at 64 for testing! Change to 2048 for true win.
+var winningTileValue = '2048';
+
 $(document).ready(function() {
-  console.log('ready!');
-  $('body').keydown(function(event){
-    var arrow_keys = [37, 38, 39, 40];
-    if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
-      event.preventDefault();
+  initializeGame();
+  $('body').keydown(playTurn);
+  $('.retry-button').click(retryClickHandler);
+});
+
+function initializeGame() {
+  // randomly pick two positions and start values
+  addTile(newTileValue);
+  addTile(newTileValue);
+}
+
+function playTurn(event) {
+  if (gameOver) { return; }
+
+  var arrow_keys = [37, 38, 39, 40];
+  if (arrow_keys.indexOf(event.which) > -1) {
+    event.preventDefault();
+
+    var collection = new TileCollection(event.which);
+    collection.move();
+
+    if ($('.tile').length >= boardSize) {
+      if (checkLose()) { endGame('lose'); }
+    } else {
+      addTile(newTileValue);
     }
-  })
-})
-
-function moveTile(tile, direction) {
-  var new_tile_value = tile.attr("data-val") * 2;
-  tile.attr("data-val", new_tile_value);
-  tile.text(new_tile_value);
-
-  switch(direction) {
-    case 38: //up
-      tile.attr("data-row","r0");
-      break;
-    case 40: //down
-      tile.attr("data-row","r3");
-      break;
-    case 37: //left
-      tile.attr("data-col","c0");
-      break;
-    case 39: //right
-      tile.attr("data-col","c3");
-      break;
+
+    if (!priorWin) { checkWin(); }
+  }
+}
+
+function retryClickHandler(event) {
+  event.preventDefault();
+  location.reload();
+}
+
+function continueClickHandler(event) {
+  event.preventDefault();
+  gameOver = false;
+  $('.gameOver, .cont-button, .retry-button').remove();
+}
+
+function addTile(possible_values) {
+  var tile = $('<div class="tile" data-row="" data-col="" data-val="">');
+  var randomRow = function() { return Math.floor(Math.random() * (rows.length)); }();
+  var randomCol = function() { return Math.floor(Math.random() * (cols.length)); }();
+  var randomValue = function() {
```

Comment:
Since you're using the randomX vars as values instead of functions, you could simply assign the value instead of using function wrappers:

```
var randomRow = Math.floor(Math.random() * (rows.length));
var randomCol = Math.floor(Math.random() * (cols.length));
var randomValue = Math.floor(Math.random() * (possible_values.length));
```